libdnf/utils/temp.hpp

```diff
@@ -0,0 +1,32 @@
+#ifndef LIBDNF_UTILS_TEMP_HPP
+#define LIBDNF_UTILS_TEMP_HPP
+
+
+#include <filesystem>
+#include <vector>
+#include <string>
+
+
+namespace libdnf::utils {
+
+
+/// Object that creates and holds a temp directory.
+/// The directory gets removed when the object is deleted.
+class TempDir {
+public:
+    explicit TempDir(const std::string & prefix);
```

Comment:
A class for a temporary directory is a good idea. But does it have to be tightly bound to the system temporary directory path?
In general, we may want to create a temporary directory anywhere. Any idea how to add this possibility?
Just an idea:
```
//Empty parent_dir_path means to use system temporary directory.
explicit TempDir(const std::string & parent_dir_path, const std::string & prefix);
```

Reply:
Ok. Forget about it. It is internal. We can change it when needed.

Reply:
That's exactly what I thought when I was writing the class. It's meant to remain internal.